lib/builder.zsh

```diff
@@ -125,22 +135,24 @@ EOF
 		dnsstr=$(echo ${output} | awk '{print $2;}')
 		ver=$(echo ${dnsstr} | sed 's,|, ,g' | awk '{print $2;}')
 
-		echo ${dnsstr} > ${tmpfile}
-		chmod 744 ${tmpfile}
-
-		publish_user=$(jq -r ".builds[${i}].publish.user" ${config})
-		publish_host=$(jq -r ".builds[${i}].publish.host" ${config})
-		publish_path=$(jq -r ".builds[${i}].publish.directory" ${config})
-		dnsentry=$(jq -r ".builds[${i}].dns" ${config})
-		apikey=$(jq -r ".apikey" ${config})
-
-		sudo -u ${publish_user} scp /builds/updater/output/update-${ver}.tar \
-		    ${publish_host}:${publish_path}/
-		sudo -u ${publish_user} scp ${tmpfile} \
-		    ${publish_host}:${publish_path}/update-latest.txt
+		publish=$(jq -r ".builds[${i}].publish" ${config})
+		if [ "${publish}" != "null" ]; then
+			do_publish ${config} ${i} ${dnsstr} ${ver}
+		fi
 
-		${TOPDIR}/updatedns.zsh ${apikey} hardenedbsd.org ${dnsentry} ${dnsstr}
+		sign=$(jq -r ".builds[${i}].sign" ${config})
+		if [ "${sign}" != "null" ]; then
```

Comment:
`.builds[N].sign` ought to be a called something more appropriate, perhaps `updatedns`. It should also be a boolean type, and checked as an optional boolean in code.